pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -454,38 +563,91 @@ async def run(self):
+
                 for record in repomd.records:
-                    checksum_types[record.type] = getattr(
-                        CHECKSUM_TYPES, record.checksum_type.upper()
-                    )
-                    record.checksum_type = getattr(CHECKSUM_TYPES, record.checksum_type.upper())
+                    record_checksum_type = getattr(CHECKSUM_TYPES, record.checksum_type.upper())
+                    checksum_types[record.type] = record_checksum_type
+                    record.checksum_type = record_checksum_type
 
-                    if record.type in types_to_download:
-                        sanitized = urlpath_sanitize(self.remote_url, record.location_href)
-                        downloader = self.remote.get_downloader(url=sanitized)
-                        metadata_downloaders[record.type] = asyncio.ensure_future(downloader.run())
+                    if not self.mirror and record.type not in types_to_download:
+                        continue
+
+                    downloader = self.remote.get_downloader(
+                        url=urlpath_sanitize(self.remote_url, record.location_href),
+                        expected_size=record.size,
+                        expected_digests={record_checksum_type: record.checksum},
+                    )
+                    repomd_downloaders[record.type] = asyncio.ensure_future(
+                        run_repomdrecord_download(record.type, record.location_href, downloader)
+                    )
 
                 self.repository.original_checksum_types = checksum_types
 
                 try:
-                    for future in asyncio.as_completed(list(metadata_downloaders.values())):
-                        await future
+                    for future in asyncio.as_completed(list(repomd_downloaders.values())):
+                        name, location_href, result = await future
+                        store_metadata_for_mirroring(self.repository, result, location_href)
+                        repomd_files[name] = result
                         metadata_pb.increment()
                 except ClientResponseError as exc:
                     raise HTTPNotFound(reason=_("File not found: {}".format(exc.request_info.url)))
 
-                metadata_files = {key: val.result() for key, val in metadata_downloaders.items()}
-                await self.parse_repository_metadata(repomd, metadata_files)
+                if self.mirror:
+                    try:
+                        downloader = self.remote.get_downloader(
+                            url=urlpath_sanitize(self.remote_url, "repodata/repomd.xml.asc")
+                        )
+                        result = await downloader.run()
+                        store_metadata_for_mirroring(
+                            self.repository, result, "repodata/repomd.xml.asc"
+                        )
+                        metadata_pb.increment()
+                    except ClientResponseError:
```

Comment:
Does it mean that if there is some network error, we'll just skip it?
Is it possible to check specifically for 404? 

If it's for any errors with code > 400, I do not think we want silently ignore it unless it's 404.
https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.raise_for_status

Thoughts?

Reply:
I'm not sure tbh.  This ties into the discussion here: https://github.com/pulp/pulp_rpm/pull/2008

Reply:
Sure, let's have it as a part of #2008, whatever we come up with there.